pkg/controller/cronjob/cronjob_controller.go

```diff
@@ -148,6 +149,23 @@ func SyncOne(sj batch.CronJob, js []batch.Job, now time.Time, jc jobControlInter
 			recorder.Eventf(&sj, v1.EventTypeNormal, "SawCompletedJob", "Saw completed job: %v", j.Name)
 		}
 	}
+
+	// Remove any job reference from the active list if the corresponding job does not exist any more.
+	// Otherwise, the cronjob may be stuck in active mode forever even though there is no matching
+	// job running.
+
+	childrenJobs := make(map[types.UID]bool)
+	for _, childJob := range js {
+		childrenJobs[childJob.ObjectMeta.UID] = true
+	}
+	for _, j := range sj.Status.Active {
+		found := childrenJobs[j.UID]
+		if !found {
+			recorder.Eventf(&sj, v1.EventTypeNormal, "MissingJob", "Active job went missing: %v", j.Name)
+			deleteFromActiveList(&sj, j.UID)
+		}
+	}
+
```

Comment:
Right, sorry my bad. Anyway, can you tweak the code a bit so that it reads nicer. Iow. move the first loop to be part of the previous one. We are already iterating through `js` there so just create `childrenJobs` map inside it. Secondly, shorten the `if` inside the second loop into:
```go
if found := childrenJobs[j.UID]; !found {
    fmt.Println("not found")
}
```